<?xml version="1.0" ?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>rsem-calculate-expression</title>
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<link rev="made" href="mailto:root@osx316.apple.com" />
</head>

<body style="background-color: white">



<ul id="index">
  <li><a href="#NAME">NAME</a></li>
  <li><a href="#SYNOPSIS">SYNOPSIS</a></li>
  <li><a href="#ARGUMENTS">ARGUMENTS</a></li>
  <li><a href="#BASIC-OPTIONS">BASIC OPTIONS</a></li>
  <li><a href="#OUTPUT-OPTIONS">OUTPUT OPTIONS</a></li>
  <li><a href="#ALIGNER-OPTIONS">ALIGNER OPTIONS</a></li>
  <li><a href="#ADVANCED-OPTIONS">ADVANCED OPTIONS</a></li>
  <li><a href="#PRIOR-ENHANCED-RSEM-OPTIONS">PRIOR-ENHANCED RSEM OPTIONS</a></li>
  <li><a href="#DEPRECATED-OPTIONS">DEPRECATED OPTIONS</a></li>
  <li><a href="#DESCRIPTION">DESCRIPTION</a></li>
  <li><a href="#NOTES">NOTES</a></li>
  <li><a href="#OUTPUT">OUTPUT</a></li>
  <li><a href="#EXAMPLES">EXAMPLES</a></li>
</ul>

<h1 id="NAME">NAME</h1>

<p>rsem-calculate-expression - Estimate gene and isoform expression from RNA-Seq data.</p>

<h1 id="SYNOPSIS">SYNOPSIS</h1>

<pre><code> rsem-calculate-expression [options] upstream_read_file(s) reference_name sample_name 
 rsem-calculate-expression [options] --paired-end upstream_read_file(s) downstream_read_file(s) reference_name sample_name 
 rsem-calculate-expression [options] --alignments [--paired-end] input reference_name sample_name</code></pre>

<h1 id="ARGUMENTS">ARGUMENTS</h1>

<dl>

<dt id="upstream_read_files-s"><b>upstream_read_files(s)</b></dt>
<dd>

<p>Comma-separated list of files containing single-end reads or upstream reads for paired-end data. By default, these files are assumed to be in FASTQ format. If the --no-qualities option is specified, then FASTA format is expected.</p>

</dd>
<dt id="downstream_read_file-s"><b>downstream_read_file(s)</b></dt>
<dd>

<p>Comma-separated list of files containing downstream reads which are paired with the upstream reads. By default, these files are assumed to be in FASTQ format. If the --no-qualities option is specified, then FASTA format is expected.</p>

</dd>
<dt id="input"><b>input</b></dt>
<dd>

<p>SAM/BAM/CRAM formatted input file. If &quot;-&quot; is specified for the filename, the input is instead assumed to come from standard input. RSEM requires all alignments of the same read group together. For paired-end reads, RSEM also requires the two mates of any alignment be adjacent. In addition, RSEM does not allow the SEQ and QUAL fields to be empty. See Description section for how to make input file obey RSEM&#39;s requirements.</p>

</dd>
<dt id="reference_name"><b>reference_name</b></dt>
<dd>

<p>The name of the reference used. The user must have run &#39;rsem-prepare-reference&#39; with this reference_name before running this program.</p>

</dd>
<dt id="sample_name"><b>sample_name</b></dt>
<dd>

<p>The name of the sample analyzed. All output files are prefixed by this name (e.g., sample_name.genes.results)</p>

</dd>
</dl>

<h1 id="BASIC-OPTIONS">BASIC OPTIONS</h1>

<dl>

<dt id="paired-end"><b>--paired-end</b></dt>
<dd>

<p>Input reads are paired-end reads. (Default: off)</p>

</dd>
<dt id="no-qualities"><b>--no-qualities</b></dt>
<dd>

<p>Input reads do not contain quality scores. (Default: off)</p>

</dd>
<dt id="strandedness-none-forward-reverse"><b>--strandedness</b> &lt;none|forward|reverse&gt;</dt>
<dd>

<p>This option defines the strandedness of the RNA-Seq reads. It recognizes three values: &#39;none&#39;, &#39;forward&#39;, and &#39;reverse&#39;. &#39;none&#39; refers to non-strand-specific protocols. &#39;forward&#39; means all (upstream) reads are derived from the forward strand. &#39;reverse&#39; means all (upstream) reads are derived from the reverse strand. If &#39;forward&#39;/&#39;reverse&#39; is set, the &#39;--norc&#39;/&#39;--nofw&#39; Bowtie/Bowtie 2 option will also be enabled to avoid aligning reads to the opposite strand. For Illumina TruSeq Stranded protocols, please use &#39;reverse&#39;. (Default: &#39;none&#39;)</p>

</dd>
<dt id="p---num-threads-int"><b>-p/--num-threads</b> &lt;int&gt;</dt>
<dd>

<p>Number of threads to use. Both Bowtie/Bowtie2, expression estimation and &#39;samtools sort&#39; will use this many threads. (Default: 1)</p>

</dd>
<dt id="alignments"><b>--alignments</b></dt>
<dd>

<p>Input file contains alignments in SAM/BAM/CRAM format. The exact file format will be determined automatically. (Default: off)</p>

</dd>
<dt id="fai-file"><b>--fai</b> &lt;file&gt;</dt>
<dd>

<p>If the header section of input alignment file does not contain reference sequence information, this option should be turned on. &lt;file&gt; is a FAI format file containing each reference sequence&#39;s name and length. Please refer to the SAM official website for the details of FAI format. (Default: off)</p>

</dd>
<dt id="bowtie2"><b>--bowtie2</b></dt>
<dd>

<p>Use Bowtie 2 instead of Bowtie to align reads. Since currently RSEM does not handle indel, local and discordant alignments, the Bowtie2 parameters are set in a way to avoid those alignments. In particular, we use options &#39;--sensitive --dpad 0 --gbar 99999999 --mp 1,1 --np 1 --score-min L,0,-0.1&#39; by default. The last parameter of &#39;--score-min&#39;, &#39;-0.1&#39;, is the negative of maximum mismatch rate. This rate can be set by option &#39;--bowtie2-mismatch-rate&#39;. If reads are paired-end, we additionally use options &#39;--no-mixed&#39; and &#39;--no-discordant&#39;. (Default: off)</p>

</dd>
<dt id="star"><b>--star</b></dt>
<dd>

<p>Use STAR to align reads. Alignment parameters are from ENCODE3&#39;s STAR-RSEM pipeline. To save computational time and memory resources, STAR&#39;s Output BAM file is unsorted. It is stored in RSEM&#39;s temporary directory with name as &#39;sample_name.bam&#39;. Each STAR job will have its own private copy of the genome in memory. (Default: off)</p>

</dd>
<dt id="hisat2-hca"><b>--hisat2-hca</b></dt>
<dd>

<p>Use HISAT2 to align reads to the transcriptome according to Human Cell Atlast SMART-Seq2 pipeline. In particular, we use HISAT parameters &quot;-k 10 --secondary --rg-id=$sampleToken --rg SM:$sampleToken --rg LB:$sampleToken --rg PL:ILLUMINA --rg PU:$sampleToken --new-summary --summary-file $sampleName.log --met-file $sampleName.hisat2.met.txt --met 5 --mp 1,1 --np 1 --score-min L,0,-0.1 --rdg 99999999,99999999 --rfg 99999999,99999999 --no-spliced-alignment --no-softclip --seed 12345&quot;. If inputs are paired-end reads, we additionally use parameters &quot;--no-mixed --no-discordant&quot;. (Default: off)</p>

</dd>
<dt id="append-names"><b>--append-names</b></dt>
<dd>

<p>If gene_name/transcript_name is available, append it to the end of gene_id/transcript_id (separated by &#39;_&#39;) in files &#39;sample_name.isoforms.results&#39; and &#39;sample_name.genes.results&#39;. (Default: off)</p>

</dd>
<dt id="seed-uint32"><b>--seed</b> &lt;uint32&gt;</dt>
<dd>

<p>Set the seed for the random number generators used in calculating posterior mean estimates and credibility intervals. The seed must be a non-negative 32 bit integer. (Default: off)</p>

</dd>
<dt id="single-cell-prior"><b>--single-cell-prior</b></dt>
<dd>

<p>By default, RSEM uses Dirichlet(1) as the prior to calculate posterior mean estimates and credibility intervals. However, much less genes are expressed in single cell RNA-Seq data. Thus, if you want to compute posterior mean estimates and/or credibility intervals and you have single-cell RNA-Seq data, you are recommended to turn on this option. Then RSEM will use Dirichlet(0.1) as the prior which encourage the sparsity of the expression levels. (Default: off)</p>

</dd>
<dt id="calc-pme"><b>--calc-pme</b></dt>
<dd>

<p>Run RSEM&#39;s collapsed Gibbs sampler to calculate posterior mean estimates. (Default: off)</p>

</dd>
<dt id="calc-ci"><b>--calc-ci</b></dt>
<dd>

<p>Calculate 95% credibility intervals and posterior mean estimates. The credibility level can be changed by setting &#39;--ci-credibility-level&#39;. (Default: off)</p>

</dd>
<dt id="q---quiet"><b>-q/--quiet</b></dt>
<dd>

<p>Suppress the output of logging information. (Default: off)</p>

</dd>
<dt id="h---help"><b>-h/--help</b></dt>
<dd>

<p>Show help information.</p>

</dd>
<dt id="version"><b>--version</b></dt>
<dd>

<p>Show version information.</p>

</dd>
</dl>

<h1 id="OUTPUT-OPTIONS">OUTPUT OPTIONS</h1>

<dl>

<dt id="sort-bam-by-read-name"><b>--sort-bam-by-read-name</b></dt>
<dd>

<p>Sort BAM file aligned under transcript coordidate by read name. Setting this option on will produce deterministic maximum likelihood estimations from independent runs. Note that sorting will take long time and lots of memory. (Default: off)</p>

</dd>
<dt id="no-bam-output"><b>--no-bam-output</b></dt>
<dd>

<p>Do not output any BAM file. (Default: off)</p>

</dd>
<dt id="sampling-for-bam"><b>--sampling-for-bam</b></dt>
<dd>

<p>When RSEM generates a BAM file, instead of outputting all alignments a read has with their posterior probabilities, one alignment is sampled according to the posterior probabilities. The sampling procedure includes the alignment to the &quot;noise&quot; transcript, which does not appear in the BAM file. Only the sampled alignment has a weight of 1. All other alignments have weight 0. If the &quot;noise&quot; transcript is sampled, all alignments appeared in the BAM file should have weight 0. (Default: off)</p>

</dd>
<dt id="output-genome-bam"><b>--output-genome-bam</b></dt>
<dd>

<p>Generate a BAM file, &#39;sample_name.genome.bam&#39;, with alignments mapped to genomic coordinates and annotated with their posterior probabilities. In addition, RSEM will call samtools (included in RSEM package) to sort and index the bam file. &#39;sample_name.genome.sorted.bam&#39; and &#39;sample_name.genome.sorted.bam.bai&#39; will be generated. (Default: off)</p>

</dd>
<dt id="sort-bam-by-coordinate"><b>--sort-bam-by-coordinate</b></dt>
<dd>

<p>Sort RSEM generated transcript and genome BAM files by coordinates and build associated indices. (Default: off)</p>

</dd>
<dt id="sort-bam-memory-per-thread-string"><b>--sort-bam-memory-per-thread</b> &lt;string&gt;</dt>
<dd>

<p>Set the maximum memory per thread that can be used by &#39;samtools sort&#39;. &lt;string&gt; represents the memory and accepts suffices &#39;K/M/G&#39;. RSEM will pass &lt;string&gt; to the &#39;-m&#39; option of &#39;samtools sort&#39;. Note that the default used here is different from the default used by samtools. (Default: 1G)</p>

</dd>
</dl>

<h1 id="ALIGNER-OPTIONS">ALIGNER OPTIONS</h1>

<dl>

<dt id="seed-length-int"><b>--seed-length</b> &lt;int&gt;</dt>
<dd>

<p>Seed length used by the read aligner. Providing the correct value is important for RSEM. If RSEM runs Bowtie, it uses this value for Bowtie&#39;s seed length parameter. Any read with its or at least one of its mates&#39; (for paired-end reads) length less than this value will be ignored. If the references are not added poly(A) tails, the minimum allowed value is 5, otherwise, the minimum allowed value is 25. Note that this script will only check if the value &gt;= 5 and give a warning message if the value &lt; 25 but &gt;= 5. (Default: 25)</p>

</dd>
<dt id="phred33-quals"><b>--phred33-quals</b></dt>
<dd>

<p>Input quality scores are encoded as Phred+33. This option is used by Bowtie, Bowtie 2 and HISAT2. (Default: on)</p>

</dd>
<dt id="phred64-quals"><b>--phred64-quals</b></dt>
<dd>

<p>Input quality scores are encoded as Phred+64 (default for GA Pipeline ver. &gt;= 1.3). This option is used by Bowtie, Bowtie 2 and HISAT2. (Default: off)</p>

</dd>
<dt id="solexa-quals"><b>--solexa-quals</b></dt>
<dd>

<p>Input quality scores are solexa encoded (from GA Pipeline ver. &lt; 1.3). This option is used by Bowtie, Bowtie 2 and HISAT2. (Default: off)</p>

</dd>
<dt id="bowtie-path-path"><b>--bowtie-path</b> &lt;path&gt;</dt>
<dd>

<p>The path to the Bowtie executables. (Default: the path to the Bowtie executables is assumed to be in the user&#39;s PATH environment variable)</p>

</dd>
<dt id="bowtie-n-int"><b>--bowtie-n</b> &lt;int&gt;</dt>
<dd>

<p>(Bowtie parameter) max # of mismatches in the seed. (Range: 0-3, Default: 2)</p>

</dd>
<dt id="bowtie-e-int"><b>--bowtie-e</b> &lt;int&gt;</dt>
<dd>

<p>(Bowtie parameter) max sum of mismatch quality scores across the alignment. (Default: 99999999)</p>

</dd>
<dt id="bowtie-m-int"><b>--bowtie-m</b> &lt;int&gt;</dt>
<dd>

<p>(Bowtie parameter) suppress all alignments for a read if &gt; &lt;int&gt; valid alignments exist. (Default: 200)</p>

</dd>
<dt id="bowtie-chunkmbs-int"><b>--bowtie-chunkmbs</b> &lt;int&gt;</dt>
<dd>

<p>(Bowtie parameter) memory allocated for best first alignment calculation (Default: 0 - use Bowtie&#39;s default)</p>

</dd>
<dt id="bowtie2-path-path"><b>--bowtie2-path</b> &lt;path&gt;</dt>
<dd>

<p>(Bowtie 2 parameter) The path to the Bowtie 2 executables. (Default: the path to the Bowtie 2 executables is assumed to be in the user&#39;s PATH environment variable)</p>

</dd>
<dt id="bowtie2-mismatch-rate-double"><b>--bowtie2-mismatch-rate</b> &lt;double&gt;</dt>
<dd>

<p>(Bowtie 2 parameter) The maximum mismatch rate allowed. (Default: 0.1)</p>

</dd>
<dt id="bowtie2-k-int"><b>--bowtie2-k</b> &lt;int&gt;</dt>
<dd>

<p>(Bowtie 2 parameter) Find up to &lt;int&gt; alignments per read. (Default: 200)</p>

</dd>
<dt id="bowtie2-sensitivity-level-string"><b>--bowtie2-sensitivity-level</b> &lt;string&gt;</dt>
<dd>

<p>(Bowtie 2 parameter) Set Bowtie 2&#39;s preset options in --end-to-end mode. This option controls how hard Bowtie 2 tries to find alignments. &lt;string&gt; must be one of &quot;very_fast&quot;, &quot;fast&quot;, &quot;sensitive&quot; and &quot;very_sensitive&quot;. The four candidates correspond to Bowtie 2&#39;s &quot;--very-fast&quot;, &quot;--fast&quot;, &quot;--sensitive&quot; and &quot;--very-sensitive&quot; options. (Default: &quot;sensitive&quot; - use Bowtie 2&#39;s default)</p>

</dd>
<dt id="star-path-path"><b>--star-path</b> &lt;path&gt;</dt>
<dd>

<p>The path to STAR&#39;s executable. (Default: the path to STAR executable is assumed to be in user&#39;s PATH environment variable)</p>

</dd>
<dt id="star-gzipped-read-file"><b>--star-gzipped-read-file</b></dt>
<dd>

<p>(STAR parameter) Input read file(s) is compressed by gzip. (Default: off)</p>

</dd>
<dt id="star-bzipped-read-file"><b>--star-bzipped-read-file</b></dt>
<dd>

<p>(STAR parameter) Input read file(s) is compressed by bzip2. (Default: off)</p>

</dd>
<dt id="star-output-genome-bam"><b>--star-output-genome-bam</b></dt>
<dd>

<p>(STAR parameter) Save the BAM file from STAR alignment under genomic coordinate to &#39;sample_name.STAR.genome.bam&#39;. This file is NOT sorted by genomic coordinate. In this file, according to STAR&#39;s manual, &#39;paired ends of an alignment are always adjacent, and multiple alignments of a read are adjacent as well&#39;. (Default: off)</p>

</dd>
<dt id="hisat2-path-path"><b>--hisat2-path</b> &lt;path&gt;</dt>
<dd>

<p>The path to HISAT2&#39;s executable. (Default: the path to HISAT2 executable is assumed to be in user&#39;s PATH environment variable)</p>

</dd>
</dl>

<h1 id="ADVANCED-OPTIONS">ADVANCED OPTIONS</h1>

<dl>

<dt id="tag-string"><b>--tag</b> &lt;string&gt;</dt>
<dd>

<p>The name of the optional field used in the SAM input for identifying a read with too many valid alignments. The field should have the format &lt;tagName&gt;:i:&lt;value&gt;, where a &lt;value&gt; bigger than 0 indicates a read with too many alignments. (Default: &quot;&quot;)</p>

</dd>
<dt id="fragment-length-min-int"><b>--fragment-length-min</b> &lt;int&gt;</dt>
<dd>

<p>Minimum read/insert length allowed. This is also the value for the Bowtie/Bowtie2 -I option. (Default: 1)</p>

</dd>
<dt id="fragment-length-max-int"><b>--fragment-length-max</b> &lt;int&gt;</dt>
<dd>

<p>Maximum read/insert length allowed. This is also the value for the Bowtie/Bowtie 2 -X option. (Default: 1000)</p>

</dd>
<dt id="fragment-length-mean-double"><b>--fragment-length-mean</b> &lt;double&gt;</dt>
<dd>

<p>(single-end data only) The mean of the fragment length distribution, which is assumed to be a Gaussian. (Default: -1, which disables use of the fragment length distribution)</p>

</dd>
<dt id="fragment-length-sd-double"><b>--fragment-length-sd</b> &lt;double&gt;</dt>
<dd>

<p>(single-end data only) The standard deviation of the fragment length distribution, which is assumed to be a Gaussian. (Default: 0, which assumes that all fragments are of the same length, given by the rounded value of <b>--fragment-length-mean</b>)</p>

</dd>
<dt id="estimate-rspd"><b>--estimate-rspd</b></dt>
<dd>

<p>Set this option if you want to estimate the read start position distribution (RSPD) from data. Otherwise, RSEM will use a uniform RSPD. (Default: off)</p>

</dd>
<dt id="num-rspd-bins-int"><b>--num-rspd-bins</b> &lt;int&gt;</dt>
<dd>

<p>Number of bins in the RSPD. Only relevant when &#39;--estimate-rspd&#39; is specified. Use of the default setting is recommended. (Default: 20)</p>

</dd>
<dt id="gibbs-burnin-int"><b>--gibbs-burnin</b> &lt;int&gt;</dt>
<dd>

<p>The number of burn-in rounds for RSEM&#39;s Gibbs sampler. Each round passes over the entire data set once. If RSEM can use multiple threads, multiple Gibbs samplers will start at the same time and all samplers share the same burn-in number. (Default: 200)</p>

</dd>
<dt id="gibbs-number-of-samples-int"><b>--gibbs-number-of-samples</b> &lt;int&gt;</dt>
<dd>

<p>The total number of count vectors RSEM will collect from its Gibbs samplers. (Default: 1000)</p>

</dd>
<dt id="gibbs-sampling-gap-int"><b>--gibbs-sampling-gap</b> &lt;int&gt;</dt>
<dd>

<p>The number of rounds between two succinct count vectors RSEM collects. If the count vector after round N is collected, the count vector after round N + &lt;int&gt; will also be collected. (Default: 1)</p>

</dd>
<dt id="ci-credibility-level-double"><b>--ci-credibility-level</b> &lt;double&gt;</dt>
<dd>

<p>The credibility level for credibility intervals. (Default: 0.95)</p>

</dd>
<dt id="ci-memory-int"><b>--ci-memory</b> &lt;int&gt;</dt>
<dd>

<p>Maximum size (in memory, MB) of the auxiliary buffer used for computing credibility intervals (CI). (Default: 1024)</p>

</dd>
<dt id="ci-number-of-samples-per-count-vector-int"><b>--ci-number-of-samples-per-count-vector</b> &lt;int&gt;</dt>
<dd>

<p>The number of read generating probability vectors sampled per sampled count vector. The crebility intervals are calculated by first sampling P(C | D) and then sampling P(Theta | C) for each sampled count vector. This option controls how many Theta vectors are sampled per sampled count vector. (Default: 50)</p>

</dd>
<dt id="keep-intermediate-files"><b>--keep-intermediate-files</b></dt>
<dd>

<p>Keep temporary files generated by RSEM. RSEM creates a temporary directory, &#39;sample_name.temp&#39;, into which it puts all intermediate output files. If this directory already exists, RSEM overwrites all files generated by previous RSEM runs inside of it. By default, after RSEM finishes, the temporary directory is deleted. Set this option to prevent the deletion of this directory and the intermediate files inside of it. (Default: off)</p>

</dd>
<dt id="temporary-folder-string"><b>--temporary-folder</b> &lt;string&gt;</dt>
<dd>

<p>Set where to put the temporary files generated by RSEM. If the folder specified does not exist, RSEM will try to create it. (Default: sample_name.temp)</p>

</dd>
<dt id="time"><b>--time</b></dt>
<dd>

<p>Output time consumed by each step of RSEM to &#39;sample_name.time&#39;. (Default: off)</p>

</dd>
</dl>

<h1 id="PRIOR-ENHANCED-RSEM-OPTIONS">PRIOR-ENHANCED RSEM OPTIONS</h1>

<dl>

<dt id="run-pRSEM"><b>--run-pRSEM</b></dt>
<dd>

<p>Running prior-enhanced RSEM (pRSEM). Prior parameters, i.e. isoform&#39;s initial pseudo-count for RSEM&#39;s Gibbs sampling, will be learned from input RNA-seq data and an external data set. When pRSEM needs and only needs ChIP-seq peak information to partition isoforms (e.g. in pRSEM&#39;s default partition model), either ChIP-seq peak file (with the &#39;--chipseq-peak-file&#39; option) or ChIP-seq FASTQ files for target and input and the path for Bowtie executables are required (with the &#39;--chipseq-target-read-files &lt;string&gt;&#39;, &#39;--chipseq-control-read-files &lt;string&gt;&#39;, and &#39;--bowtie-path &lt;path&gt; options), otherwise, ChIP-seq FASTQ files for target and control and the path to Bowtie executables are required. (Default: off)</p>

</dd>
<dt id="chipseq-peak-file-string"><b>--chipseq-peak-file</b> &lt;string&gt;</dt>
<dd>

<p>Full path to a ChIP-seq peak file in ENCODE&#39;s narrowPeak, i.e. BED6+4, format. This file is used when running prior-enhanced RSEM in the default two-partition model. It partitions isoforms by whether they have ChIP-seq overlapping with their transcription start site region or not. Each partition will have its own prior parameter learned from a training set. This file can be either gzipped or ungzipped. (Default: &quot;&quot;)</p>

</dd>
<dt id="chipseq-target-read-files-string"><b>--chipseq-target-read-files</b> &lt;string&gt;</dt>
<dd>

<p>Comma-separated full path of FASTQ read file(s) for ChIP-seq target. This option is used when running prior-enhanced RSEM. It provides information to calculate ChIP-seq peaks and signals. The file(s) can be either ungzipped or gzipped with a suffix &#39;.gz&#39; or &#39;.gzip&#39;. The options &#39;--bowtie-path &lt;path&gt;&#39; and &#39;--chipseq-control-read-files &lt;string&gt;&#39; must be defined when this option is specified. (Default: &quot;&quot;)</p>

</dd>
<dt id="chipseq-control-read-files-string"><b>--chipseq-control-read-files</b> &lt;string&gt;</dt>
<dd>

<p>Comma-separated full path of FASTQ read file(s) for ChIP-seq conrol. This option is used when running prior-enhanced RSEM. It provides information to call ChIP-seq peaks. The file(s) can be either ungzipped or gzipped with a suffix &#39;.gz&#39; or &#39;.gzip&#39;. The options &#39;--bowtie-path &lt;path&gt;&#39; and &#39;--chipseq-target-read-files &lt;string&gt;&#39; must be defined when this option is specified. (Default: &quot;&quot;)</p>

</dd>
<dt id="chipseq-read-files-multi-targets-string"><b>--chipseq-read-files-multi-targets</b> &lt;string&gt;</dt>
<dd>

<p>Comma-separated full path of FASTQ read files for multiple ChIP-seq targets. This option is used when running prior-enhanced RSEM, where prior is learned from multiple complementary data sets. It provides information to calculate ChIP-seq signals. All files can be either ungzipped or gzipped with a suffix &#39;.gz&#39; or &#39;.gzip&#39;. When this option is specified, the option &#39;--bowtie-path &lt;path&gt;&#39; must be defined and the option &#39;--partition-model &lt;string&gt;&#39; will be set to &#39;cmb_lgt&#39; automatically. (Default: &quot;&quot;)</p>

</dd>
<dt id="chipseq-bed-files-multi-targets-string"><b>--chipseq-bed-files-multi-targets</b> &lt;string&gt;</dt>
<dd>

<p>Comma-separated full path of BED files for multiple ChIP-seq targets. This option is used when running prior-enhanced RSEM, where prior is learned from multiple complementary data sets. It provides information of ChIP-seq signals and must have at least the first six BED columns. All files can be either ungzipped or gzipped with a suffix &#39;.gz&#39; or &#39;.gzip&#39;. When this option is specified, the option &#39;--partition-model &lt;string&gt;&#39; will be set to &#39;cmb_lgt&#39; automatically. (Default: &quot;&quot;)</p>

</dd>
<dt id="cap-stacked-chipseq-reads"><b>--cap-stacked-chipseq-reads</b></dt>
<dd>

<p>Keep a maximum number of ChIP-seq reads that aligned to the same genomic interval. This option is used when running prior-enhanced RSEM, where prior is learned from multiple complementary data sets. This option is only in use when either &#39;--chipseq-read-files-multi-targets &lt;string&gt;&#39; or &#39;--chipseq-bed-files-multi-targets &lt;string&gt;&#39; is specified. (Default: off)</p>

</dd>
<dt id="n-max-stacked-chipseq-reads-int"><b>--n-max-stacked-chipseq-reads</b> &lt;int&gt;</dt>
<dd>

<p>The maximum number of stacked ChIP-seq reads to keep. This option is used when running prior-enhanced RSEM, where prior is learned from multiple complementary data sets. This option is only in use when the option &#39;--cap-stacked-chipseq-reads&#39; is set. (Default: 5)</p>

</dd>
<dt id="partition-model-string"><b>--partition-model</b> &lt;string&gt;</dt>
<dd>

<p>A keyword to specify the partition model used by prior-enhanced RSEM. It must be one of the following keywords:</p>

<dl>

<dt id="pk">- <b>pk</b></dt>
<dd>

<p>Partitioned by whether an isoform has a ChIP-seq peak overlapping with its transcription start site (TSS) region. The TSS region is defined as [TSS-500bp, TSS+500bp]. For simplicity, we refer this type of peak as &#39;TSS peak&#39; when explaining other keywords.</p>

</dd>
<dt id="pk_lgtnopk">- <b>pk_lgtnopk</b></dt>
<dd>

<p>First partitioned by TSS peak. Then, for isoforms in the &#39;no TSS peak&#39; set, a logistic model is employed to further classify them into two partitions.</p>

</dd>
<dt id="lm3-lm4-lm5-or-lm6">- <b>lm3</b>, <b>lm4</b>, <b>lm5</b>, or <b>lm6</b></dt>
<dd>

<p>Based on their ChIP-seq signals, isoforms are classified into 3, 4, 5, or 6 partitions by a linear regression model.</p>

</dd>
<dt id="nopk_lm2pk-nopk_lm3pk-nopk_lm4pk-or-nopk_lm5pk">- <b>nopk_lm2pk</b>, <b>nopk_lm3pk</b>, <b>nopk_lm4pk</b>, or <b>nopk_lm5pk</b></dt>
<dd>

<p>First partitioned by TSS peak. Then, for isoforms in the &#39;with TSS peak&#39; set, a linear regression model is employed to further classify them into 2, 3, 4, or 5 partitions.</p>

</dd>
<dt id="pk_lm2nopk-pk_lm3nopk-pk_lm4nopk-or-pk_lm5nopk">- <b>pk_lm2nopk</b>, <b>pk_lm3nopk</b>, <b>pk_lm4nopk</b>, or <b>pk_lm5nopk</b></dt>
<dd>

<p>First partitioned by TSS peak. Then, for isoforms in the &#39;no TSS peak&#39; set, a linear regression model is employed to further classify them into 2, 3, 4, or 5 partitions.</p>

</dd>
<dt id="cmb_lgt">- <b>cmb_lgt</b></dt>
<dd>

<p>Using a logistic regression to combine TSS signals from multiple complementary data sets and partition training set isoform into &#39;expressed&#39; and &#39;not expressed&#39;. This partition model is only in use when either &#39;--chipseq-read-files-multi-targets &lt;string&gt;&#39; or &#39;--chipseq-bed-files-multi-targets &lt;string&gt; is specified.</p>

</dd>
</dl>

<p>Parameters for all the above models are learned from a training set. For detailed explanations, please see prior-enhanced RSEM&#39;s paper. (Default: &#39;pk&#39;)</p>

</dd>
</dl>

<h1 id="DEPRECATED-OPTIONS">DEPRECATED OPTIONS</h1>

<ul>

<p>The options in this section are deprecated. They are here only for compatibility reasons and may be removed in future releases.</p>

</ul>

<dl>

<dt id="sam"><b>--sam</b></dt>
<dd>

<p>Inputs are alignments in SAM format. (Default: off)</p>

</dd>
<dt id="bam"><b>--bam</b></dt>
<dd>

<p>Inputs are alignments in BAM format. (Default: off)</p>

</dd>
<dt id="strand-specific"><b>--strand-specific</b></dt>
<dd>

<p>Equivalent to &#39;--strandedness forward&#39;. (Default: off)</p>

</dd>
<dt id="forward-prob-double"><b>--forward-prob</b> &lt;double&gt;</dt>
<dd>

<p>Probability of generating a read from the forward strand of a transcript. Set to 1 for a strand-specific protocol where all (upstream) reads are derived from the forward strand, 0 for a strand-specific protocol where all (upstream) read are derived from the reverse strand, or 0.5 for a non-strand-specific protocol. (Default: off)</p>

</dd>
</dl>

<h1 id="DESCRIPTION">DESCRIPTION</h1>

<p>In its default mode, this program aligns input reads against a reference transcriptome with Bowtie and calculates expression values using the alignments. RSEM assumes the data are single-end reads with quality scores, unless the &#39;--paired-end&#39; or &#39;--no-qualities&#39; options are specified. Alternatively, users can use STAR to align reads using the &#39;--star&#39; option. RSEM has provided options in &#39;rsem-prepare-reference&#39; to prepare STAR&#39;s genome indices. Users may use an alternative aligner by specifying &#39;--alignments&#39;, and providing an alignment file in SAM/BAM/CRAM format. However, users should make sure that they align against the indices generated by &#39;rsem-prepare-reference&#39; and the alignment file satisfies the requirements mentioned in ARGUMENTS section.</p>

<p>One simple way to make the alignment file satisfying RSEM&#39;s requirements is to use the &#39;convert-sam-for-rsem&#39; script. This script accepts SAM/BAM/CRAM files as input and outputs a BAM file. For example, type the following command to convert a SAM file, &#39;input.sam&#39;, to a ready-for-use BAM file, &#39;input_for_rsem.bam&#39;:</p>

<pre><code>  convert-sam-for-rsem input.sam input_for_rsem  </code></pre>

<p>For details, please refer to &#39;convert-sam-for-rsem&#39;s documentation page.</p>

<h1 id="NOTES">NOTES</h1>

<p>1. Users must run &#39;rsem-prepare-reference&#39; with the appropriate reference before using this program.</p>

<p>2. For single-end data, it is strongly recommended that the user provide the fragment length distribution parameters (--fragment-length-mean and --fragment-length-sd). For paired-end data, RSEM will automatically learn a fragment length distribution from the data.</p>

<p>3. Some aligner parameters have default values different from their original settings.</p>

<p>4. With the &#39;--calc-pme&#39; option, posterior mean estimates will be calculated in addition to maximum likelihood estimates.</p>

<p>5. With the &#39;--calc-ci&#39; option, 95% credibility intervals and posterior mean estimates will be calculated in addition to maximum likelihood estimates.</p>

<p>6. The temporary directory and all intermediate files will be removed when RSEM finishes unless &#39;--keep-intermediate-files&#39; is specified.</p>

<p>With the &#39;--run-pRSEM&#39; option and associated options (see section &#39;PRIOR-ENHANCED RSEM OPTIONS&#39; above for details), prior-enhanced RSEM will be running. Prior parameters will be learned from supplied external data set(s) and assigned as initial pseudo-counts for isoforms in the corresponding partition for Gibbs sampling.</p>

<h1 id="OUTPUT">OUTPUT</h1>

<dl>

<dt id="sample_name.isoforms.results"><b>sample_name.isoforms.results</b></dt>
<dd>

<p>File containing isoform level expression estimates. The first line contains column names separated by the tab character. The format of each line in the rest of this file is:</p>

<p>transcript_id gene_id length effective_length expected_count TPM FPKM IsoPct [posterior_mean_count posterior_standard_deviation_of_count pme_TPM pme_FPKM IsoPct_from_pme_TPM TPM_ci_lower_bound TPM_ci_upper_bound TPM_coefficient_of_quartile_variation FPKM_ci_lower_bound FPKM_ci_upper_bound FPKM_coefficient_of_quartile_variation]</p>

<p>Fields are separated by the tab character. Fields within &quot;[]&quot; are optional. They will not be presented if neither &#39;--calc-pme&#39; nor &#39;--calc-ci&#39; is set.</p>

<p>&#39;transcript_id&#39; is the transcript name of this transcript. &#39;gene_id&#39; is the gene name of the gene which this transcript belongs to (denote this gene as its parent gene). If no gene information is provided, &#39;gene_id&#39; and &#39;transcript_id&#39; are the same.</p>

<p>&#39;length&#39; is this transcript&#39;s sequence length (poly(A) tail is not counted). &#39;effective_length&#39; counts only the positions that can generate a valid fragment. If no poly(A) tail is added, &#39;effective_length&#39; is equal to transcript length - mean fragment length + 1. If one transcript&#39;s effective length is less than 1, this transcript&#39;s both effective length and abundance estimates are set to 0.</p>

<p>&#39;expected_count&#39; is the sum of the posterior probability of each read comes from this transcript over all reads. Because 1) each read aligning to this transcript has a probability of being generated from background noise; 2) RSEM may filter some alignable low quality reads, the sum of expected counts for all transcript are generally less than the total number of reads aligned.</p>

<p>&#39;TPM&#39; stands for Transcripts Per Million. It is a relative measure of transcript abundance. The sum of all transcripts&#39; TPM is 1 million. &#39;FPKM&#39; stands for Fragments Per Kilobase of transcript per Million mapped reads. It is another relative measure of transcript abundance. If we define l_bar be the mean transcript length in a sample, which can be calculated as</p>

<p>l_bar = \sum_i TPM_i / 10^6 * effective_length_i (i goes through every transcript),</p>

<p>the following equation is hold:</p>

<p>FPKM_i = 10^3 / l_bar * TPM_i.</p>

<p>We can see that the sum of FPKM is not a constant across samples.</p>

<p>&#39;IsoPct&#39; stands for isoform percentage. It is the percentage of this transcript&#39;s abandunce over its parent gene&#39;s abandunce. If its parent gene has only one isoform or the gene information is not provided, this field will be set to 100.</p>

<p>&#39;posterior_mean_count&#39;, &#39;pme_TPM&#39;, &#39;pme_FPKM&#39; are posterior mean estimates calculated by RSEM&#39;s Gibbs sampler. &#39;posterior_standard_deviation_of_count&#39; is the posterior standard deviation of counts. &#39;IsoPct_from_pme_TPM&#39; is the isoform percentage calculated from &#39;pme_TPM&#39; values.</p>

<p>&#39;TPM_ci_lower_bound&#39;, &#39;TPM_ci_upper_bound&#39;, &#39;FPKM_ci_lower_bound&#39; and &#39;FPKM_ci_upper_bound&#39; are lower(l) and upper(u) bounds of 95% credibility intervals for TPM and FPKM values. The bounds are inclusive (i.e. [l, u]).</p>

<p>&#39;TPM_coefficient_of_quartile_variation&#39; and &#39;FPKM_coefficient_of_quartile_variation&#39; are coefficients of quartile variation (CQV) for TPM and FPKM values. CQV is a robust way of measuring the ratio between the standard deviation and the mean. It is defined as</p>

<p>CQV := (Q3 - Q1) / (Q3 + Q1),</p>

<p>where Q1 and Q3 are the first and third quartiles.</p>

</dd>
<dt id="sample_name.genes.results"><b>sample_name.genes.results</b></dt>
<dd>

<p>File containing gene level expression estimates. The first line contains column names separated by the tab character. The format of each line in the rest of this file is:</p>

<p>gene_id transcript_id(s) length effective_length expected_count TPM FPKM [posterior_mean_count posterior_standard_deviation_of_count pme_TPM pme_FPKM TPM_ci_lower_bound TPM_ci_upper_bound TPM_coefficient_of_quartile_variation FPKM_ci_lower_bound FPKM_ci_upper_bound FPKM_coefficient_of_quartile_variation]</p>

<p>Fields are separated by the tab character. Fields within &quot;[]&quot; are optional. They will not be presented if neither &#39;--calc-pme&#39; nor &#39;--calc-ci&#39; is set.</p>

<p>&#39;transcript_id(s)&#39; is a comma-separated list of transcript_ids belonging to this gene. If no gene information is provided, &#39;gene_id&#39; and &#39;transcript_id(s)&#39; are identical (the &#39;transcript_id&#39;).</p>

<p>A gene&#39;s &#39;length&#39; and &#39;effective_length&#39; are defined as the weighted average of its transcripts&#39; lengths and effective lengths (weighted by &#39;IsoPct&#39;). A gene&#39;s abundance estimates are just the sum of its transcripts&#39; abundance estimates.</p>

</dd>
<dt id="sample_name.alleles.results"><b>sample_name.alleles.results</b></dt>
<dd>

<p>Only generated when the RSEM references are built with allele-specific transcripts.</p>

<p>This file contains allele level expression estimates for allele-specific expression calculation. The first line contains column names separated by the tab character. The format of each line in the rest of this file is:</p>

<p>allele_id transcript_id gene_id length effective_length expected_count TPM FPKM AlleleIsoPct AlleleGenePct [posterior_mean_count posterior_standard_deviation_of_count pme_TPM pme_FPKM AlleleIsoPct_from_pme_TPM AlleleGenePct_from_pme_TPM TPM_ci_lower_bound TPM_ci_upper_bound TPM_coefficient_of_quartile_variation FPKM_ci_lower_bound FPKM_ci_upper_bound FPKM_coefficient_of_quartile_variation]</p>

<p>Fields are separated by the tab character. Fields within &quot;[]&quot; are optional. They will not be presented if neither &#39;--calc-pme&#39; nor &#39;--calc-ci&#39; is set.</p>

<p>&#39;allele_id&#39; is the allele-specific name of this allele-specific transcript.</p>

<p>&#39;AlleleIsoPct&#39; stands for allele-specific percentage on isoform level. It is the percentage of this allele-specific transcript&#39;s abundance over its parent transcript&#39;s abundance. If its parent transcript has only one allele variant form, this field will be set to 100.</p>

<p>&#39;AlleleGenePct&#39; stands for allele-specific percentage on gene level. It is the percentage of this allele-specific transcript&#39;s abundance over its parent gene&#39;s abundance.</p>

<p>&#39;AlleleIsoPct_from_pme_TPM&#39; and &#39;AlleleGenePct_from_pme_TPM&#39; have similar meanings. They are calculated based on posterior mean estimates.</p>

<p>Please note that if this file is present, the fields &#39;length&#39; and &#39;effective_length&#39; in &#39;sample_name.isoforms.results&#39; should be interpreted similarly as the corresponding definitions in &#39;sample_name.genes.results&#39;.</p>

</dd>
<dt id="sample_name.transcript.bam"><b>sample_name.transcript.bam</b></dt>
<dd>

<p>Only generated when --no-bam-output is not specified.</p>

<p>&#39;sample_name.transcript.bam&#39; is a BAM-formatted file of read alignments in transcript coordinates. The MAPQ field of each alignment is set to min(100, floor(-10 * log10(1.0 - w) + 0.5)), where w is the posterior probability of that alignment being the true mapping of a read. In addition, RSEM pads a new tag ZW:f:value, where value is a single precision floating number representing the posterior probability. Because this file contains all alignment lines produced by bowtie or user-specified aligners, it can also be used as a replacement of the aligner generated BAM/SAM file.</p>

</dd>
<dt id="sample_name.transcript.sorted.bam-and-sample_name.transcript.sorted.bam.bai"><b>sample_name.transcript.sorted.bam and sample_name.transcript.sorted.bam.bai</b></dt>
<dd>

<p>Only generated when --no-bam-output is not specified and --sort-bam-by-coordinate is specified.</p>

<p>&#39;sample_name.transcript.sorted.bam&#39; and &#39;sample_name.transcript.sorted.bam.bai&#39; are the sorted BAM file and indices generated by samtools (included in RSEM package).</p>

</dd>
<dt id="sample_name.genome.bam"><b>sample_name.genome.bam</b></dt>
<dd>

<p>Only generated when --no-bam-output is not specified and --output-genome-bam is specified.</p>

<p>&#39;sample_name.genome.bam&#39; is a BAM-formatted file of read alignments in genomic coordinates. Alignments of reads that have identical genomic coordinates (i.e., alignments to different isoforms that share the same genomic region) are collapsed into one alignment. The MAPQ field of each alignment is set to min(100, floor(-10 * log10(1.0 - w) + 0.5)), where w is the posterior probability of that alignment being the true mapping of a read. In addition, RSEM pads a new tag ZW:f:value, where value is a single precision floating number representing the posterior probability. If an alignment is spliced, a XS:A:value tag is also added, where value is either &#39;+&#39; or &#39;-&#39; indicating the strand of the transcript it aligns to.</p>

</dd>
<dt id="sample_name.genome.sorted.bam-and-sample_name.genome.sorted.bam.bai"><b>sample_name.genome.sorted.bam and sample_name.genome.sorted.bam.bai</b></dt>
<dd>

<p>Only generated when --no-bam-output is not specified, and --sort-bam-by-coordinate and --output-genome-bam are specified.</p>

<p>&#39;sample_name.genome.sorted.bam&#39; and &#39;sample_name.genome.sorted.bam.bai&#39; are the sorted BAM file and indices generated by samtools (included in RSEM package).</p>

</dd>
<dt id="sample_name.time"><b>sample_name.time</b></dt>
<dd>

<p>Only generated when --time is specified.</p>

<p>It contains time (in seconds) consumed by aligning reads, estimating expression levels and calculating credibility intervals.</p>

</dd>
<dt id="sample_name.log"><b>sample_name.log</b></dt>
<dd>

<p>Only generated when --alignments is not specified.</p>

<p>It captures alignment statistics outputted from the user-specified aligner.</p>

</dd>
<dt id="sample_name.stat"><b>sample_name.stat</b></dt>
<dd>

<p>This is a folder instead of a file. All model related statistics are stored in this folder. Use &#39;rsem-plot-model&#39; can generate plots using this folder.</p>

<p>&#39;sample_name.stat/sample_name.cnt&#39; contains alignment statistics. The format and meanings of each field are described in &#39;cnt_file_description.txt&#39; under RSEM directory.</p>

<p>&#39;sample_name.stat/sample_name.model&#39; stores RNA-Seq model parameters learned from the data. The format and meanings of each filed of this file are described in &#39;model_file_description.txt&#39; under RSEM directory.</p>

<p>The following four output files will be generated only by prior-enhanced RSEM</p>

<dl>

<dt id="sample_name.stat-sample_name_prsem.all_tr_features">- &#39;sample_name.stat/sample_name_prsem.all_tr_features&#39;</dt>
<dd>

<p>It stores isofrom features for deriving and assigning pRSEM prior. The first line is a header and the rest is one isoform per line. The description for each column is:</p>

<ul>

<li><p><b>trid</b>: transcript ID from input annotation</p>

</li>
<li><p><b>geneid</b>: gene ID from input anntation</p>

</li>
<li><p><b>chrom</b>: isoform&#39;s chromosome name</p>

</li>
<li><p><b>strand</b>: isoform&#39;s strand name</p>

</li>
<li><p><b>start</b>: isoform&#39;s end with the lowest genomic loci</p>

</li>
<li><p><b>end</b>: isoform&#39;s end with the highest genomic loci</p>

</li>
<li><p><b>tss_mpp</b>: average mappability of [TSS-500bp, TSS+500bp], where TSS is isoform&#39;s transcription start site, i.e. 5&#39;-end</p>

</li>
<li><p><b>body_mpp</b>: average mappability of (TSS+500bp, TES-500bp), where TES is isoform&#39;s transcription end site, i.e. 3&#39;-end</p>

</li>
<li><p><b>tes_mpp</b>: average mappability of [TES-500bp, TES+500bp]</p>

</li>
<li><p><b>pme_count</b>: isoform&#39;s fragment or read count from RSEM&#39;s posterior mean estimates</p>

</li>
<li><p><b>tss</b>: isoform&#39;s TSS loci</p>

</li>
<li><p><b>tss_pk</b>: equal to 1 if isoform&#39;s [TSS-500bp, TSS+500bp] region overlaps with a RNA Pol II peak; 0 otherwise</p>

</li>
<li><p><b>is_training</b>: equal to 1 if isoform is in the training set where Pol II prior is learned; 0 otherwise</p>

</li>
</ul>

</dd>
<dt id="sample_name.stat-sample_name_prsem.all_tr_prior">- &#39;sample_name.stat/sample_name_prsem.all_tr_prior&#39;</dt>
<dd>

<p>It stores prior parameters for every isoform. This file does not have a header. Each line contains a prior parameter and an isoform&#39;s transcript ID delimited by ` # `.</p>

</dd>
<dt id="sample_name.stat-sample_name_uniform_prior_1.isoforms.results">- &#39;sample_name.stat/sample_name_uniform_prior_1.isoforms.results&#39;</dt>
<dd>

<p>RSEM&#39;s posterior mean estimates on the isoform level with an initial pseudo-count of one for every isoform. It is in the same format as the &#39;sample_name.isoforms.results&#39;.</p>

</dd>
<dt id="sample_name.stat-sample_name_uniform_prior_1.genes.results">- &#39;sample_name.stat/sample_name_uniform_prior_1.genes.results&#39;</dt>
<dd>

<p>RSEM&#39;s posterior mean estimates on the gene level with an initial pseudo-count of one for every isoform. It is in the same format as the &#39;sample_name.genes.results&#39;.</p>

</dd>
</dl>

<p>When learning prior from multiple external data sets in prior-enhanced RSEM, two additional output files will be generated.</p>

<dl>

<dt id="sample_name.stat-sample_name.pval_LL">- &#39;sample_name.stat/sample_name.pval_LL&#39;</dt>
<dd>

<p>It stores a p-value and a log-likelihood. The p-value indicates whether the combination of multiple complementary data sets is informative for RNA-seq quantification. The log-likelihood shows how well pRSEM&#39;s Dirichlet-multinomial model fits the read counts of partitioned training set isoforms.</p>

</dd>
<dt id="sample_name.stat-sample_name.lgt_mdl.RData">- &#39;sample_name.stat/sample_name.lgt_mdl.RData&#39;</dt>
<dd>

<p>It stores an R object named &#39;glmmdl&#39;, which is a logistic regression model on the training set isoforms and multiple external data sets.</p>

</dd>
</dl>

<p>In addition, extra columns will be added to &#39;sample_name.stat/all_tr_features&#39;</p>

<ul>

<li><p><b>is_expr</b>: equal to 1 if isoform has an abundance &gt;= 1 TPM and a non-zero read count from RSEM&#39;s posterior mean estimates; 0 otherwise</p>

</li>
<li><p><b>&quot;$external_data_set_basename&quot;</b>: log10 of external data&#39;s signal at [TSS-500, TSS+500]. Signal is the number of reads aligned within that interval and normalized to RPKM by read depth and interval length. It will be set to -4 if no read aligned to that interval.</p>

<p>There are multiple columns like this one, where each represents an external data set.</p>

</li>
<li><p><b>prd_expr_prob</b>: predicted probability from logistic regression model on whether this isoform is expressed or not. A probability higher than 0.5 is considered as expressed</p>

</li>
<li><p><b>partition</b>: group index, to which this isoforms is partitioned</p>

</li>
<li><p><b>prior</b>: prior parameter for this isoform</p>

</li>
</ul>

</dd>
</dl>

<h1 id="EXAMPLES">EXAMPLES</h1>

<p>Assume the path to the bowtie executables is in the user&#39;s PATH environment variable. Reference files are under &#39;/ref&#39; with name &#39;mouse_125&#39;.</p>

<p>1) &#39;/data/mmliver.fq&#39;, single-end reads with quality scores. Quality scores are encoded as for &#39;GA pipeline version &gt;= 1.3&#39;. We want to use 8 threads and generate a genome BAM file. In addition, we want to append gene/transcript names to the result files:</p>

<pre><code> rsem-calculate-expression --phred64-quals \
                           -p 8 \
                           --append-names \
                           --output-genome-bam \
                           /data/mmliver.fq \
                           /ref/mouse_125 \
                           mmliver_single_quals</code></pre>

<p>2) &#39;/data/mmliver_1.fq&#39; and &#39;/data/mmliver_2.fq&#39;, stranded paired-end reads with quality scores. Suppose the library is prepared using TruSeq Stranded Kit, which means the first mate should map to the reverse strand. Quality scores are in SANGER format. We want to use 8 threads and do not generate a genome BAM file:</p>

<pre><code> rsem-calculate-expression -p 8 \
                           --paired-end \
                           --strandedness reverse \
                           /data/mmliver_1.fq \
                           /data/mmliver_2.fq \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>

<p>3) &#39;/data/mmliver.fa&#39;, single-end reads without quality scores. We want to use 8 threads:</p>

<pre><code> rsem-calculate-expression -p 8 \
                           --no-qualities \
                           /data/mmliver.fa \
                           /ref/mouse_125 \
                           mmliver_single_without_quals</code></pre>

<p>4) Data are the same as 1). This time we assume the bowtie executables are under &#39;/sw/bowtie&#39;. We want to take a fragment length distribution into consideration. We set the fragment length mean to 150 and the standard deviation to 35. In addition to a BAM file, we also want to generate credibility intervals. We allow RSEM to use 1GB of memory for CI calculation:</p>

<pre><code> rsem-calculate-expression --bowtie-path /sw/bowtie \
                           --phred64-quals \
                           --fragment-length-mean 150.0 \
                           --fragment-length-sd 35.0 \
                           -p 8 \
                           --output-genome-bam \
                           --calc-ci \
                           --ci-memory 1024 \
                           /data/mmliver.fq \
                           /ref/mouse_125 \
                           mmliver_single_quals</code></pre>

<p>5) &#39;/data/mmliver_paired_end_quals.bam&#39;, BAM-formatted alignments for paired-end reads with quality scores. We want to use 8 threads:</p>

<pre><code> rsem-calculate-expression --paired-end \
                           --alignments \
                           -p 8 \
                           /data/mmliver_paired_end_quals.bam \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>

<p>6) &#39;/data/mmliver_1.fq.gz&#39; and &#39;/data/mmliver_2.fq.gz&#39;, paired-end reads with quality scores and read files are compressed by gzip. We want to use STAR to aligned reads and assume STAR executable is &#39;/sw/STAR&#39;. Suppose we want to use 8 threads and do not generate a genome BAM file:</p>

<pre><code> rsem-calculate-expression --paired-end \
                           --star \
                           --star-path /sw/STAR \
                           --gzipped-read-file \
                           --paired-end \
                           -p 8 \
                           /data/mmliver_1.fq.gz \
                           /data/mmliver_2.fq.gz \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>

<p>7) In the above example, suppose we want to run prior-enhanced RSEM instead. Assuming we want to learn priors from a ChIP-seq peak file &#39;/data/mmlive.narrowPeak.gz&#39;:</p>

<pre><code> rsem-calculate-expression --star \
                           --star-path /sw/STAR \
                           --gzipped-read-file \
                           --paired-end \
                           --calc-pme \
                           --run-pRSEM \
                           --chipseq-peak-file /data/mmliver.narrowPeak.gz \
                           -p 8 \
                           /data/mmliver_1.fq.gz \
                           /data/mmliver_2.fq.gz \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>

<p>8) Similar to the example in 7), suppose we want to use the partition model &#39;pk_lm2nopk&#39; (partitioning isoforms by Pol II TSS peak first and then partitioning &#39;no TSS peak&#39; isoforms into two bins by a linear regression model), and we want to partition isoforms by RNA Pol II&#39;s ChIP-seq read files &#39;/data/mmliver_PolIIRep1.fq.gz&#39; and &#39;/data/mmliver_PolIIRep2.fq.gz&#39;, and the control ChIP-seq read files &#39;/data/mmliver_ChIPseqCtrl.fq.gz&#39;. Also, assuming Bowtie&#39;s executables are under &#39;/sw/bowtie/&#39;:</p>

<pre><code> rsem-calculate-expression --star \
                           --star-path /sw/STAR \
                           --gzipped-read-file \
                           --paired-end \
                           --calc-pme \
                           --run-pRSEM \
                           --chipseq-target-read-files /data/mmliver_PolIIRep1.fq.gz,/data/mmliver_PolIIRep2.fq.gz \
                           --chipseq-control-read-files /data/mmliver_ChIPseqCtrl.fq.gz \
                           --partition-model pk_lm2nopk \
                           --bowtie-path /sw/bowtie \
                           -p 8 \
                           /data/mmliver_1.fq.gz \
                           /data/mmliver_2.fq.gz \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>

<p>9) Similar to the example in 8), suppose we want to derive prior from four histone modification ChIP-seq read data sets: &#39;/data/H3K27Ac.fastq.gz&#39;, &#39;/data/H3K4me1.fastq.gz&#39;, &#39;/data/H3K4me2.fastq.gz&#39;, and &#39;/data/H3K4me3.fastq.gz&#39;. Also, assuming Bowtie&#39;s executables are under &#39;/sw/bowtie/&#39;:</p>

<pre><code> rsem-calculate-expression --star \
                           --star-path /sw/STAR \
                           --gzipped-read-file \
                           --paired-end \
                           --calc-pme \
                           --run-pRSEM \
                           --partition-model cmb_lgt \
                           --chipseq-read-files-multi-targets /data/H3K27Ac.fastq.gz,/data/H3K4me1.fastq.gz,/data/H3K4me2.fastq.gz,/data/H3K4me3.fastq.gz \
                           --bowtie-path /sw/bowtie \
                           -p 8 \
                           /data/mmliver_1.fq.gz \
                           /data/mmliver_2.fq.gz \
                           /ref/mouse_125 \
                           mmliver_paired_end_quals</code></pre>


</body>

</html>


